### PR TITLE
DATAES-135 Adapt to new multi-store configuration detection and reposito...

### PR DESCRIPTION
**Pull request by kevinleturc**
...ries extending ElasticsearchRepository are now considered strict matches.

Here is my PR.

Hope its help.

### CODE REVIEW

File: src/main/java/org/springframework/data/elasticsearch/repository/config/ElasticsearchRepositoryConfigExtension.java [hunk 1]
```diff
@@ -51,7 +58,33 @@ protected String getModulePrefix() {
 		return "elasticsearch";
 	}
 
-	/* 
+	/*
+ 	 * (non-Javadoc)
+	 * @see org.springframework.data.repository.config.RepositoryConfigurationExtensionSupport#getModuleName()
+	 */
+	@Override
+	public String getModuleName() {
```
Comment (olivergierke) [inline src/main/java/org/springframework/data/elasticsearch/repository/config/ElasticsearchRepositoryConfigExtension.java]: This method is not needed as it basically returns what's the default anyway.

File: src/main/java/org/springframework/data/elasticsearch/repository/config/EnableElasticsearchRepositories.java [hunk 2]
```diff
@@ -111,4 +112,10 @@
 	 * @return
 	 */
 	String elasticsearchTemplateRef() default "elasticsearchTemplate";
+
+	/**
+	 * Configures whether nested repository-interfaces (e.g. defined as inner classes) should be discovered by the
+	 * repositories infrastructure.
+	 */
+	boolean considerNestedRepositories() default false;
```
Comment (olivergierke) [inline src/main/java/org/springframework/data/elasticsearch/repository/config/EnableElasticsearchRepositories.java]: We should move this into a separate pull request alongside dedicated tests as it's got nothing to do with the original ticket.

File: src/main/java/org/springframework/data/elasticsearch/repository/config/ElasticsearchRepositoryConfigExtension.java (hunk 1 above)
Comment (kevinleturc) [inline src/main/java/org/springframework/data/elasticsearch/repository/config/ElasticsearchRepositoryConfigExtension.java]: There's no case issues on this ? If not I remove it in next commit.

File: src/main/java/org/springframework/data/elasticsearch/repository/config/EnableElasticsearchRepositories.java (hunk 2 above)
Comment (kevinleturc) [inline src/main/java/org/springframework/data/elasticsearch/repository/config/EnableElasticsearchRepositories.java]: Ok, I did it in this PR https://github.com/spring-projects/spring-data-elasticsearch/pull/91

**Comment by kevinleturc** [comment]
Hi, I did the asked changements.
Thanks

**Comment by kevinleturc** [comment]
Hi,
Any news on this ?
Regards.

**Comment by akonczak** [comment]
Thanks Kevin for your pull request. I have picked up all the changes regarding nested repositories. 
The multi-store part has been done by DATA-136
